### PR TITLE
Amoled Theme (Pure Black Theme)

### PR DESCRIPTION
**Pull request by Rajala1404**
I think it would be pretty nice to Implement an Amoled Theme (Pure Black Background) because on AMOLED displays pure black looks really good (at least in my opinion). I already have a implementation for this Feature that still Supports Material You Color Theming (except for the Background of course). Some People argue that Pure Black is bad because it makes for them it more difficult to read the Text but that shouldn't be a Problem because the Small Text are in there Chips that are Themed by Material You. My Implementation is still not Final because the App needs to be restart if you switch to AMOLED or back from AMOLED but that can be fixed by a Popup that Gives a Button to Restart or something like that. After I'm done i would open a Pull Request. Here are some Pictures of my Implementation, just need Feedback if that would be good: 
combined1
combined2
combined3
The Background of Timetable Popup is not Pure Black that just looks like that on the Screenshot or maybe its just my Crappy TV Screen

### CODE REVIEW

**Comment by Rajala1404** [comment]
Feature implementation is done. Switching to Amoled and away from Amoled needs an Restart from the App (doesn't work if Dart Compiler is attached, App closes then but works in Flawless without Dart Compiler attached) switching between Light, Dark, System still happens dynamically (so no restart needed)
combined

File: app/lib/themes.dart [hunk 1]
```diff
@@ -10,14 +12,43 @@ class Themes {
...
   static Themes getNewTheme(Color seedColor) {
-    // The basic theme, global theme data changes should be put here.
     ThemeData basicTheme(Brightness brightness) {
-      return ThemeData(
-        colorScheme:
-            ColorScheme.fromSeed(seedColor: seedColor, brightness: brightness),
-        inputDecorationTheme:
-            const InputDecorationTheme(border: OutlineInputBorder()),
-      );
+      if (globalStorage.prefs.getString("theme") == "amoled") {
```
Comment (alessioC42) [inline app/lib/themes.dart:16]: We do not use magic strings to use the globalStorage class. Have a look at storage.dart and other calls of globalStorage

**Review by alessioC42 — COMMENTED** [review]
I cannot get my head around why a restart is required at all when there should only be a change in the theme. Could you explain why it is required?

**Review by alessioC42 — CHANGES_REQUESTED** [review]
Also, the code contains a lot of "special cases" for amoled (which leads to unreadability), mostly due to the restart implementation.

**Comment by Rajala1404** [comment]
> I cannot get my head around why a restart is required at all when there should only be a change in the theme. Could you explain why it is required?

The Restart is required because i can't find a way to reload the ThemeData or just change the needed Values outside of ThemeData and if the App doesn't get restarted an it attemps Dynamic switching it causes unintended behavior. If you dynamicly change from Dark to Amoled, nothing happens, or if you dynamicly change from Amoled to Light, it looks really wierd

**Comment by alessioC42** [comment]
https://stackoverflow.com/questions/49164592/flutter-how-to-change-the-materialapp-theme-at-runtime
This approach should be modifiable to work. Introducing enums or a custom data class for themes with accent color or something and using the streambuilder to update it. Even the current DynamicColorBuilder should be integratable. 

I defently won't merge the "restart" approach due to its unnatural UX design element.

**Comment by Rajala1404** [comment]
> https://stackoverflow.com/questions/49164592/flutter-how-to-change-the-materialapp-theme-at-runtime
>
Yeah that looks good i try this approach

**Comment by kurwjan** [comment]
Couldn't you just add "amoled" to the `ThemeModeNotifier`? I'll test it.

**Comment by kurwjan** [comment]
``` dart
return ValueListenableBuilder<Themes>(
          valueListenable: ColorModeNotifier.notifier,
          builder: (_, theme, __) {
            return ValueListenableBuilder<ThemeMode>(
                valueListenable: ThemeModeNotifier.notifier,
                builder: (_, mode, __) {
                  return MaterialApp(
                    title: 'Lanis Mobile',
                    theme: theme.lightTheme,
                    darkTheme: theme.darkTheme,
                    themeMode: mode,
                    localizationsDelegates:
                        AppLocalizations.localizationsDelegates,
                    supportedLocales: AppLocalizations.supportedLocales,
                    home: const StartupScreen(),
                  );
                });
          });
```

You could check here if a ValueNotifier Amoled is true with ValueListenableBuilder, make a function which takes the current dark theme (`theme.darkTheme`) and return it with `theme.darkTheme.copyWith(your modification....)` in `themes.dart`.

**Comment by Rajala1404** [comment]
@kurwjan You're approach sounds good but if i try to implement it its the same problem as before that it doesn't get changed until a reload (maybe i just misintepreterted your idea if so just say it) i tried it like this:
```dart
amoled(Themes theme) {
      // Colors for Amoled Mode
      final Map<String, Color> amoledColors = {
        "background": Colors.black,
        "secondary": const Color(0xFF0f0f0f),
        "third": const Color(0xFF0a0a0a),
      };
      String? currentTheme = globalStorage.prefs.getString("theme");

      if (currentTheme == "amoled") {
        return theme.darkTheme?.copyWith(
          // Amoled Background & Themes for required Components
            scaffoldBackgroundColor: amoledColors["background"],
            navigationBarTheme: NavigationBarThemeData(
              backgroundColor: amoledColors["background"],
            ),
            ...other modifications
        );
      } else {
        return theme.darkTheme;
      }
    }

      ...

      return ValueListenableBuilder<Themes>(
          valueListenable: ColorModeNotifier.notifier,
          builder: (_, theme, __) {
            return ValueListenableBuilder<ThemeMode>(
                valueListenable: ThemeModeNotifier.notifier,
                builder: (_, mode, __) {
                  return MaterialApp(
                    title: 'Lanis Mobile',
                    theme: theme.lightTheme,
                    darkTheme: amoled(theme),
                    themeMode: mode,
                    localizationsDelegates:
                    AppLocalizations.localizationsDelegates,
                    supportedLocales: AppLocalizations.supportedLocales,
                    home: const StartupScreen(),
                  );
                });
          });
    });
```

**Comment by kurwjan** [comment]
I added a new ValueListenableBuilder with a ValueNotifier (`AmoledNotifier`) packed in class with other useful  like the other notifiers and also changed the Amoled radio button to a checkbox (i didn't said that bc i came up with this idea later).

But code still needs to be cleaned up and translation.

**Comment by Rajala1404** [comment]
Thanks @kurwjan for the implementation, i cleaned up all my commented code (that needed the Restart) and translation is already present from my previous commits so i think it's ready for a merge. Tested with my S23+ (Android 14) and looks beautiful.

**Comment by Rajala1404** [comment]
Ok after some further testing i noticed that the Amoled theme was not preserved after a restart (the checkbox was preserved (checked)) maybe it doesn't get properly loaded at startup it starts working again after unchecking and rechecking the box

**Comment by Rajala1404** [comment]
Fixed that amoled mode was not preserved after restart. 
At 123 instead of calling `String isAmoled = await globalStorage.read(key: StorageKey.settingsIsAmoled);` for the isAmoled setting it called `String theme =await globalStorage.read(key: StorageKey.settingsSelectedTheme);` which caused it to return `dark` instead of true/false

**Review by alessioC42 — APPROVED** [review]
Looks fine for me. @kurwjan any thoughts?

**Review by kurwjan — APPROVED** [review]
super

**Pull request merged**